packages/eslint-plugin/tests/rules/keyword-spacing.test.ts

```diff
@@ -0,0 +1,4168 @@
+    {
+      code: '(() => typeof foo)',
+      options: [NEITHER],
+      parserOptions: { ecmaVersion: 6 },
+    },
+
+    // not conflict with `block-spacing`
+    '{typeof foo }',
+    { code: '{ typeof foo }', options: [NEITHER] },
+
+    // not conflict with `comma-spacing`
+    '(0,typeof foo)',
+    { code: '(0, typeof foo)', options: [NEITHER] },
+
+    // not conflict with `computed-property-spacing`
+    'a[typeof foo]',
+    { code: '({[typeof foo]: 0})', parserOptions: { ecmaVersion: 6 } },
+    { code: 'a[ typeof foo]', options: [NEITHER] },
+    {
+      code: '({[ typeof foo]: 0})',
+      options: [NEITHER],
+      parserOptions: { ecmaVersion: 6 },
+    },
+
+    // not conflict with `key-spacing`
+    '({a:typeof foo })',
+    { code: '({a: typeof foo })', options: [NEITHER] },
+
+    // not conflict with `semi-spacing`
+    ';typeof foo',
+    { code: '; typeof foo', options: [NEITHER] },
+
+    // not conflict with `space-in-parens`
+    '(typeof foo)',
+    { code: '( typeof foo)', options: [NEITHER] },
+
+    // not conflict with `space-infix-ops`
+    'a =typeof foo',
+    { code: 'a = typeof foo', options: [NEITHER] },
+
+    // not conflict with `space-unary-ops`
+    '!typeof+foo',
+    { code: '! typeof +foo', options: [NEITHER] },
+
+    // not conflict with `template-curly-spacing`
+    { code: '`${typeof foo}`', parserOptions: { ecmaVersion: 6 } },
+    {
+      code: '`${ typeof foo}`',
+      options: [NEITHER],
+      parserOptions: { ecmaVersion: 6 },
+    },
+
+    // not conflict with `jsx-curly-spacing`
+    {
+      code: '<Foo onClick={typeof foo} />',
+      parserOptions: { ecmaFeatures: { jsx: true } },
+    },
+    {
+      code: '<Foo onClick={ typeof foo} />',
+      options: [NEITHER],
+      parserOptions: { ecmaFeatures: { jsx: true } },
+    },
+
+    //----------------------------------------------------------------------
+    // var
+    //----------------------------------------------------------------------
+
+    { code: '{} var [a] = b', parserOptions: { ecmaVersion: 6 } },
+    {
+      code: '{}var[a] = b',
+      options: [NEITHER],
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: '{} var [a] = b',
+      options: [override('var', BOTH)],
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: '{}var[a] = b',
+      options: [override('var', NEITHER)],
+      parserOptions: { ecmaVersion: 6 },
+    },
+    'for (var foo in [1, 2, 3]) {}',
+
+    // not conflict with `block-spacing`
+    '{var a = b }',
+    { code: '{ var a = b }', options: [NEITHER] },
+
+    // not conflict with `semi-spacing`
+    ';var a = b',
+    { code: '; var a = b', options: [NEITHER] },
+
+    //----------------------------------------------------------------------
+    // void
+    //----------------------------------------------------------------------
+
+    '{} void foo',
+    { code: '{}void foo', options: [NEITHER] },
+    { code: '{} void foo', options: [over
```

Comment:
This shouldn't be needed - if it is, that's a sign that there's a type that needs to be adjusted somewhere.

(My first guess would be the return types of the helper methods)

Reply:
:/ this was meant to be on the [`as TSESLint.InvalidTestCase<MessageIds, Options>[],`](https://github.com/typescript-eslint/typescript-eslint/pull/1739/files#diff-9ec4a33b0114a94ee107a0f20c938ebeR4167).